torch/fx/experimental/symbolic_shapes.py

```diff
@@ -240,41 +254,42 @@ class FloorDiv(sympy.Function):
         """
         nargs = (2,)
 
-        def _sympystr(self, printer):
-            lhs = self.args[0]
-            rhs = self.args[1]
-            lhs_str = printer._print(lhs)
-            rhs_str = printer._print(rhs)
-            if precedence(lhs) < precedence(sympy.div):
-                lhs_str = f"({lhs_str})"
-            if precedence(rhs) < precedence(sympy.div):
-                rhs_str = f"({rhs_str})"
+        # Default return type. For instance, this applies when both arguments
+        # are Symbols without any assumptions.
+        # See NOTE [ SymPy eval and assumptions ]
+        is_real = True
+
+        @property
+        def base(self):
+            return self.args[0]
 
-            return f"{lhs_str}//{rhs_str}"
+        @property
+        def divisor(self):
+            return self.args[1]
 
+        def _sympystr(self, printer):
+            base_str = printer._print(self.base)
+            divisor_str = printer._print(self.divisor)
+            if precedence(self.base) < precedence(sympy.div):
+                base_str = f"({base_str})"
+            if precedence(self.divisor) < precedence(sympy.div):
+                divisor_str = f"({divisor_str})"
+            return f"{base_str}//{divisor_str}"
```

Comment:
Changes here are just due to me introducing properties. I think this is more readable.